### PR TITLE
refactor(deps): Upgrade peer dependencies to latest working versions

### PR DESCRIPTION
**Pull request by jstoffan**
BREAKING CHANGE: Peer dependencies have been updated and will require host applications to update as well to avoid npm warnings.

```
Bundle size comparison (before/after):
picker.js (878 KiB)	picker.js (887 KiB)
uploader.js (764 KiB)	uploader.js (774 KiB)
explorer.js (1.6 MiB)	explorer.js (1.61 MiB)
preview.js (1.36 MiB)	preview.js (1.37 MiB)
sidebar.js (1.28 MiB)	sidebar.js (1.28 MiB)
openwith.js (643 KiB)	openwith.js (652 KiB)

```

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/components/slide-carousel/README.md [hunk 1]
```diff
@@ -8,24 +8,20 @@ const Slide = require('./Slide').default;
 <div style={ { width: "700px" } }>
     <SlideCarousel contentHeight="250px" initialIndex={3} title="Love & Happiness">
         <Slide>
-            <Logo/>
```
Comment (jstoffan) [inline src/components/slide-carousel/README.md]: None of these components are being imported above, so this was throwing errors on the `/Components` page.

File: examples/src/SelectorDropdownExamples.js [hunk 2]
```diff
@@ -131,7 +131,7 @@ const SelectorDropdownExamples = () => (
             'The Chronic',
             'Straight Outta Compton',
             'Reasonable Doubt',
-            'Super long name that should be truncated, we should see the dots at the very end, adding some more text here, should truncate at any moment now, please truncate soon',
+            'Super long name that should be truncated, we should see the dots at the very end, adding some more text here, should truncate at any second now, please truncate soon',
```
Comment (ConradJChan) [inline examples/src/SelectorDropdownExamples.js]: Was this a result from search and replace of `moment`? 😄

Comment (jstoffan) [reply examples/src/SelectorDropdownExamples.js]: In a way. I got tired of seeing it in my search results, so I went and changed it manually.

File: package.json [hunk 3]
```diff
@@ -267,24 +268,24 @@
         "immutable": "^3.7.4",
         "jsuri": "^1.3.1",
         "lodash": "^4.17.5",
-        "mousetrap": "^1.6.1",
-        "pikaday": "^1.6.1",
+        "mousetrap": "^1.6.3",
```
Comment (priyajeet) [inline package.json]: Should we be making peers use newer patches?

Comment (jstoffan) [reply package.json]: Looks like I missed a few. Will push an updated list.

**Pull request merged**